tests/functional/testPushMessageHandlers.js

```diff
@@ -0,0 +1,157 @@
+          {
+            name: 'currentDevice',
+            unique_id: 'currentDevice',
+            pubkey: await vpn.getPublicKey(),
+            ipv4_address: '127.0.0.1',
+            ipv6_address: '::1',
+            created_at: new Date().toISOString()
+          },
+          {
+            name: 'device_2',
+            unique_id: 'device_2',
+            pubkey: 'publicKey_2',
+            ipv4_address: '127.0.0.1',
+            ipv6_address: '::1',
+            created_at: new Date().toISOString()
+          },
+          {
+            name: 'device_3',
+            unique_id: 'device_3',
+            pubkey: 'publicKey_3',
+            ipv4_address: '127.0.0.1',
+            ipv6_address: '::1',
+            created_at: new Date().toISOString()
+          },
+          {
+            name: 'device_4',
+            unique_id: 'device_4',
+            pubkey: 'publicKey_4',
+            ipv4_address: '127.0.0.1',
+            ipv6_address: '::1',
+            created_at: new Date().toISOString()
+          },
+          {
+            name: 'device_5',
+            unique_id: 'device_5',
+            pubkey: 'publicKey_5',
+            ipv4_address: '127.0.0.1',
+            ipv6_address: '::1',
+            created_at: new Date().toISOString()
+          }
+        ],
+      };
+
+      this.ctx.guardianOverrideEndpoints = {
+        GETs: {
+          '/api/v1/vpn/account': {status: 200, body: UserData},
+        },
+      };
+    });
+
+    it('no-op', () => {
+      // We need this test case here for due to the following situation:
+      //
+      // We need the guardian overrides to be inside a `beforeEach` block,
+      // otherwise we cannot use `await` to wait for the publicKey function to complete.
+      //
+      // That means it runs _after_ the VPN is initialized for the first time
+      // which means the overrides are only effective after a test case has already run.
+    })
+
+    it('attempting to delete the current device leads to a logout', async () => {
+      // Wait for viewMain to load, then wait a bit more for good measure.
+      await vpn.waitForElement("viewMainFlickable");
+      await vpn.wait();
+
+      const devices = await vpn.getDevices();
+      guardian.broadcastMessage(JSON.stringify({
+        type: messageType,
+        payload: {
+          publicKey: devices.find(device => device.currentDevice).publicKey,
+        }
+      }));
+
+      await vpn.waitForElementProperty('VPN', 'userState', 'UserNotAuthenticated');
+    });
+
+    it('attempting to delete a third device updates the device list', async () => {
+      // Wait for viewMain to load, then wait a bit more for good measure.
+      await vpn.waitForElement("viewMainFlickable");
+      await vpn.wait();
+
+      let devices = await vpn.getDevices();
+      // Get a device from the list of devices that is not the current one.
+      let thirdDevice = devices.find(device => !device.currentDevice);
+      while(thirdDevice) {
```

Comment:
why a loop? is it not enough 1 single broadcastMessage?

Reply:
yes please. Because it should simply work. I don't want that this "covers" other bugs. What do you think?

Reply:
Oh, I think I misunderstood your initial question. There is a loop there because we are deleting more than one device, not because we attempt to delete the same device more than once. I chose a poor variable name, `thirdDevice` makes it sound like there are only three, but in fact there are 5 devices there (see the Guardian overrides on the beaforeEach block). I'll change that.

I'd still rather leave the loop there, so that we know deletion works no matter how many devices are in the list. WDYT?

Reply:
good!